lib/ansible/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -138,18 +154,19 @@ def get_project(self, project_name):
     def list_all_project_variables(self):
         return self.project.variables.list()
 
-    def create_variable(self, key, value):
+    def create_variable(self, key, value, masked, protected):
         if self._module.check_mode:
             return
-        return self.project.variables.create({"key": key, "value": value})
+        return self.project.variables.create({"key": key, "value": value,
+                                              "masked": masked, "protected": protected})
 
-    def update_variable(self, var, value):
-        if var.value == value:
+    def update_variable(self, key, var, value, masked, protected):
+        if var.value == value and var.protected == protected and var.masked == masked:
             return False
         if self._module.check_mode:
             return True
-        var.value = value
-        var.save()
+        self.delete_variable(key)
+        self.create_variable(key, value, masked, protected)
```

Comment:
How about using `save()` when `maked` and `protected` does not change, and only use `delete_variable` + `create_variable` when these change?